lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -226,10 +226,8 @@ export class AuthorizationCodeClient extends BaseClient {
             parameterBuilder.addClientSecret(this.config.clientCredentials.clientSecret);
         }
 
-        // Use clientAssertion from request, fallback to client assertion in base configuration
-        const clientAssertion = request.clientAssertion || this.config.clientCredentials.clientAssertion;
-
-        if (clientAssertion) {
+        if (this.config.clientCredentials.clientAssertion) {
```

Comment:
This has been rolled back to the behavior before adding client assertion to the request since the auth code, OBO, refresh token and UPN flows won't use a client assertion. We should consider removing the client assertion logic from non-client credential clients in the future.